### PR TITLE
[IOTDB-951] Fix auto-creating schema in parallel may write unrecognizable timeseries in mlog.txt

### PR DESCRIPTION
**Pull request by wshao08**
Jira Issue: https://issues.apache.org/jira/browse/IOTDB-951

### CODE REVIEW

**Review by HTHou — APPROVED** [review]

**Review by JackieTien97 — CHANGES_REQUESTED** [review]
Firstly, the method need not to be synchronnized.
Secondly, the FileChannel instance could be inialized in the contruct function, and shouldn't contruct it every time in the writeTimeseries function.

File: server/src/main/java/org/apache/iotdb/db/metadata/MLogWriter.java [hunk 1]
```diff
@@ -38,6 +41,8 @@
   private static final String STRING_TYPE = "%s,%s,%s"; 
   private File logFile;
   private BufferedWriter writer;
```
Comment (JackieTien97) [inline server/src/main/java/org/apache/iotdb/db/metadata/MLogWriter.java:43]: You can use fileChannel to totally replace it.

**Review by JackieTien97 — CHANGES_REQUESTED** [review]

File: server/src/main/java/org/apache/iotdb/db/metadata/MLogWriter.java [hunk 2]
```diff
@@ -53,40 +58,50 @@ public MLogWriter(String schemaDir, String logFileName) throws IOException {
...
-    newLine();
+    final String newLine = System.getProperty("line.separator");
+    buf.append(newLine);
+
+    channel.write(ByteBuffer.wrap(buf.toString().getBytes()));
+    channel.force(true);
+    ++lineNumber;
   }
 
   public void deleteTimeseries(String path) throws IOException {
```
Comment (samperson1997) [inline server/src/main/java/org/apache/iotdb/db/metadata/MLogWriter.java:107]: I think you could use `channel.write()` to implement all the functions in this class, including `deleteTimeseries`, `setStorageGroup`, `deleteStorageGroup`, `setTTL`, `changeOffset` and so on

File: server/src/main/java/org/apache/iotdb/db/metadata/MLogWriter.java (hunk 1 above)
Comment (SilverNarcissus) [inline server/src/main/java/org/apache/iotdb/db/metadata/MLogWriter.java:43]: We can use file channel to replace it. Check other place where it is used ~

**Review by SilverNarcissus — CHANGES_REQUESTED** [review]
Nice work! Just few issues, please have a look~

File: server/src/main/java/org/apache/iotdb/db/metadata/MLogWriter.java (hunk 1 above)
Comment (wshao08) [reply server/src/main/java/org/apache/iotdb/db/metadata/MLogWriter.java]: Fixed

Comment (wshao08) [reply server/src/main/java/org/apache/iotdb/db/metadata/MLogWriter.java]: Fixed

**Review by SilverNarcissus — APPROVED** [review]
Brilliant work!  Look good to me

**Review by samperson1997 — APPROVED** [review]

**Pull request merged**